pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperKitCreationService.java

```diff
@@ -0,0 +1,276 @@
+
+        }
+
+        log.info(juniperKitRequest.getJuniperKitId() + " " + ddpInstance.getName() + " " + kitTypeName + " kit created");
+        return new JuniperKitStatus(null, null, null);
+    }
+
+    /**
+     * checkAddress tries creating an address in EasyPost. If it is successful,
+     * sets the kit's easypostAddressId and returns true, if not returns false
+     * An address is valid only if participant has shortId, first - and lastName, for Juniper shortId is the juniperParticipantId
+     *
+     * @param juniperKitRequest the JuniperKitRequest with address to check
+     */
+
+    public boolean checkAddress(JuniperKitRequest juniperKitRequest, String phone, EasyPostUtil easyPostUtil) {
+        if ((StringUtils.isBlank(juniperKitRequest.getJuniperParticipantID()))
+                || StringUtils.isBlank(juniperKitRequest.getLastName())) {
+            return false;
+        }
+        //let's validate the participant's address
+        String name = "";
+        if (StringUtils.isNotBlank(juniperKitRequest.getFirstName())) {
+            name += juniperKitRequest.getFirstName() + " ";
+        }
+        name += juniperKitRequest.getLastName();
+        if (juniperKitRequest.isSkipAddressValidation()) {
+            try {
+                Address address = easyPostUtil.createBroadAddress(name, juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(),
+                        juniperKitRequest.getCity(),
+                        juniperKitRequest.getPostalCode(), juniperKitRequest.getState(), juniperKitRequest.getCountry(), phone);
+                juniperKitRequest.setEasypostAddressId(address.getId());
+                return true;
+            } catch (EasyPostException e) {
+                log.warn("Easypost couldn't create an address for " + juniperKitRequest.getShortId());
+                e.printStackTrace();
+                return false;
+            }
+        }
+        DeliveryAddress deliveryAddress =
+                new DeliveryAddress(juniperKitRequest.getStreet1(), juniperKitRequest.getStreet2(), juniperKitRequest.getCity(),
+                        juniperKitRequest.getState(),
+                        juniperKitRequest.getPostalCode(), juniperKitRequest.getCountry(), name, phone);
+        deliveryAddress.validate();
+        if (deliveryAddress.isValid()) {
+            //store the address back
+            juniperKitRequest.setEasypostAddressId(deliveryAddress.getId());
+            return true;
+        }
+        log.info("Address is not valid " + juniperKitRequest.getShortId());
+        return false;
+
+    }
+
+
+    /**
+     * This method creates a collaborator participant id for the kit and then inserts in DB
+     *
+     * @return SimpleResult of the interaction with database
+     */
+    private SimpleResult createKit(@NonNull DDPInstance ddpInstance, @NonNull KitType kitType, JuniperKitRequest kit,
```

Comment:
The use of SimpleResult as a return type feels clumsy. We are jamming response information into a throwable type that we ultimately will not throw.
Would it make more sense to return a KitResponse?